FAQ.md

```diff
@@ -0,0 +1,290 @@
+5. Use XML if an [XML root tag is found](https://github.com/github/linguist/blob/master/lib/linguist/strategy/xml.rb).
+6. Use the language determined by a set of regexp-based [heuristic rules][`heuristics.yml`].
+7. Use the best matched language returned by a naïve Bayesian classifier trained on [sample files][].
+   This is the last strategy with the lowest accuracy.
+   The classifier always takes a subset of languages as input; it is not meant to classify all languages.
+
+
+### How does Linguist work on GitHub.com?
+
+When you push changes to a repository on GitHub.com, a low priority background job is enqueued to analyze the files in your repository as explained in [*How does Linguist detect the language of a file?*](#how-does-linguist-detect-the-language-of-a-file).
+The results of this analysis are cached for the lifetime of your repository and are only updated when the repository is updated.
+
+As this analysis is performed by a low priority background job, it can take a while, particularly during busy periods, for your language statistics bar to reflect your changes.
+
+
+### How is a repository's language determined?
+
+A common misconception is that GitHub reports a repository language.
+This is not the case.
+Instead, GitHub uses a concept of _"this repository contains these languages in X, Y and Z proportions"_, and these proportions are determined by Linguist based on the bytes of code found within the repository.
+The "repository language" you see is the most prominent language detected and this is what is is shown next to the repository's name on some pages.
+
+
+### How can I change the language of my repository?
+
+If you believe the language statistics for your repository are incorrect, please see [*The language statistics in my repository are wrong*](#the-language-statistics-in-my-repository-are-wrong).
+If you believe we missed a language or one of its extensions, please consider [submitting a pull request](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language) if said extensions meet [the requirements](#what-are-the-requirements-to-associate-a-new-extension-to-a-language).
+If everything looks correct, but you'd still like another language to appear first, please consider using an [override][].
+
+
+### The language detected for some files in my repository is incorrect.
+
+You can force Linguist to determine the correct language using an [override][].
+
+It's also likely you can help us improve things.
+If you believe we don't support a language we should, and if that language is [widespread enough](#what-are-the-requirements-to-add-support-for-a-new-language-or-associate-a-new-extension-with-a-language), you can [send us a pull request](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language).
```

Comment:
I wouldn't call these *details*. Since we're heavily relying on user's contributions, I think it's important to have little reminders in the right places that a contribution/improvement is a just a few clicks away.